addon/core/schema.ts

```diff
@@ -55,38 +55,60 @@ export function getRdfaAttrs(node: Element): RdfaAttrs | false {
   };
 
   let hasAnyRdfaAttributes = false;
-  for (const key of Object.keys(rdfaDomAttrs)) {
+  for (const key of [...Object.keys(rdfaDomAttrs), 'about']) {
     const value = node.attributes.getNamedItem(key)?.value;
     if (isSome(value)) {
       attrs[key] = value;
       hasAnyRdfaAttributes = true;
 
-      if (key === 'data-outgoing-props') {
-        const properties = JSON.parse(value) as Property[];
-        attrs['properties'] = properties;
-      }
-      if (key === 'data-incoming-props') {
-        const backlinks = JSON.parse(value) as Backlink[];
-        attrs['backlinks'] = backlinks;
-      }
-      if (key === 'data-rdfa-node-type') {
-        const type = value as unknown as RdfaAttrs['rdfaNodeType'];
-        if (!rdfaNodeTypes.includes(type)) {
-          logger('rdfaNodeType is not a valid type', value, node);
+      switch (key) {
+        case 'data-outgoing-props': {
+          const properties = JSON.parse(value) as Property[];
+          attrs['properties'] = properties;
+          break;
+        }
+        case 'data-incoming-props': {
+          const backlinks = JSON.parse(value) as Backlink[];
+          attrs['backlinks'] = backlinks;
+          break;
+        }
+        case 'data-rdfa-node-type': {
+          const type = value as unknown as RdfaAttrs['rdfaNodeType'];
+          if (!rdfaNodeTypes.includes(type)) {
+            logger('rdfaNodeType is not a valid type', value, node);
+          }
+          if (type === 'resource') {
+            attrs = {
+              ...attrs,
+              rdfaNodeType: type,
+              resource:
+                attrs.resource && typeof attrs.resource === 'string'
+                  ? attrs.resource
+                  : '',
+              properties:
+                attrs.properties && attrs.properties instanceof Array
+                  ? (attrs.properties as Property[])
+                  : [],
+            };
+          }
+          break;
         }
-        if (type === 'resource') {
```

Comment:
I know this is already merged, but I didn't notice before

so if I understand correctly this is making sure resource nodes _have_ a resource attribute, even if there is none in the dom, right?
It's a bit janky cause it relies on the ordering of the rdfaDomAttrs constant, and Object.keys looping over the keys in order (which is probably in the spec, but still it's a bit fragile)

I'd do these checks after the loop to make it a bit more clear

Reply:
ah yeah it would indeed be fine cause we don't overwrite when there's no value, so even if we hit resource later in the loop we would either overwrite it with the right value, or not do anything
I'd still put it outside of the loop personally, but its preference

Reply:
Fair. The typing is easier this way though, so I'll leave it here. As soon as I set `rdfaNodeType` to be `resource`, it insists that the others are present.